src/kbmod/region_search.py

```diff
@@ -1,424 +1,659 @@
-                uri = uri.geturl()  # Convert to URL string
-                uris.append(trim_uri_func(uri))
-            except Exception as e:
-                print(f"Failed to retrieve path for dataId {data_id}: {e}")
-        return uris
-
-    def get_uris(self, data_ids, dataset_types=None, collections=None, trim_uri_func=_trim_uri):
-        """
-        Get the URIs for the given dataIds.
+        set of int
+            The indices of the patches that contain the ephemeris entries.
+        """
+        if guess_dist is not None and guess_dist != 0.0 and guess_dist not in self.guess_dists:
+            raise ValueError(f"Guess distance {guess_dist} not specified for RegionSearch")
+        if guess_dist is None:
+            guess_dist = 0.0
+
+        # Iterate over all items of the ephemeris and check
+        # if they are in any of the patches
+        ephems_ras = ephems.get_ras(guess_dist)
+        ephems_decs = ephems.get_decs(guess_dist)
+
+        # For each ephemeris entry, check if it is in any of the patches
+        # We need to check if the ephemeris entry is in any of the patches with search around sky
+        # coordinates. We do this by checking if the ephemeris entry is in any of the patches
+        # with a search radius of half the patch size
+        ephems_coords = SkyCoord(
+            ephems_ras,
+            ephems_decs,
+            unit=(u.deg, u.deg),
+            frame="icrs",
+        )
+        # Get the patch center coordinates
+        patch_centers = SkyCoord(
+            [patch.ra for patch in self.patches],
+            [patch.dec for patch in self.patches],
+            unit=(u.deg, u.deg),
+            frame="icrs",
+        )
+
+        # Use search_around_sky to find the indices of the patches that may contain the ephemeris entries
+        # As our search radius from the patch center, we use the distance between the center of the patch
+        # and the corners of the patch, which can simply be calculated using the pythagorean theorem.
+        patch_size_deg = np.sqrt((self.patches[0].width / 2) ** 2 + (self.patches[0].height / 2) ** 2) * u.deg
+
+        # The elements of the returned lists are indices within the ephemeris and patch center coordinates
+        # that are within patch_size_deg of each other
+        ephems_idx, patch_idx, _, _ = search_around_sky(ephems_coords, patch_centers, patch_size_deg)
+
+        # Now we need to check if the ephemeris entry is actually in the patch
+        res_patch_indices = set([])
+        for ephem_idx, patch_idx in zip(ephems_idx, patch_idx):
+            if patch_idx not in res_patch_indices:
+                # Check if the ephemeris entry is in the patch
+                curr_ra, curr_dec = ephems_coords[ephem_idx].ra.deg, ephems_coords[ephem_idx].dec.deg
+                if self.patches[patch_idx].contains(curr_ra, curr_dec):
+                    res_patch_indices.add(patch_idx)
+        return res_patch_indices
```

Comment:
Just to make sure I'm understanding correctly - the user passes in an ephemeris with many ra/dec values, and this function identifies all the patch ids that contain at least one of the ra/dec values from the ephemeris.

And if the user passes in an ephemeris with ra/decs that evenly cover the sky, we would likely return all the patch ids?

If that understanding is correct, is the expected use case that the user will generally pass in only a single ra/dec for a point on the sky to search around?